osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -11,48 +11,51 @@ namespace osu.Game.Rulesets.Osu.Difficulty.Preprocessing
 {
     public class OsuDifficultyHitObject : DifficultyHitObject
     {
-        private const int normalized_radius = 50; // Change radius to 50 to make 100 the diameter. Easier for mental maths.
+        private const int normalised_radius = 50; // Change radius to 50 to make 100 the diameter. Easier for mental maths.
         private const int min_delta_time = 25;
-        private const float maximum_slider_radius = normalized_radius * 2.4f;
-        private const float assumed_slider_radius = normalized_radius * 1.8f;
+        private const float maximum_slider_radius = normalised_radius * 2.4f;
+        private const float assumed_slider_radius = normalised_radius * 1.8f;
 
         protected new OsuHitObject BaseObject => (OsuHitObject)base.BaseObject;
 
         /// <summary>
-        /// Normalized distance from the end position of the previous <see cref="OsuDifficultyHitObject"/> to the start position of this <see cref="OsuDifficultyHitObject"/>.
+        /// Milliseconds elapsed since the start time of the previous <see cref="OsuDifficultyHitObject"/>, with a minimum of 25ms.
         /// </summary>
-        public double JumpDistance { get; private set; }
+        public readonly double StrainTime;
 
         /// <summary>
-        /// Minimum distance from the end position of the previous <see cref="OsuDifficultyHitObject"/> to the start position of this <see cref="OsuDifficultyHitObject"/>.
+        /// Normalised distance from the end position of the previous <see cref="OsuDifficultyHitObject"/> to the start position of this <see cref="OsuDifficultyHitObject"/>.
         /// </summary>
-        public double MovementDistance { get; private set; }
+        public double JumpDistance { get; private set; }
 
         /// <summary>
-        /// Normalized distance between the start and end position of the previous <see cref="OsuDifficultyHitObject"/>.
+        /// Normalised minimum distance from the end position of the previous <see cref="OsuDifficultyHitObject"/> to the start position of this <see cref="OsuDifficultyHitObject"/>.
         /// </summary>
-        public double TravelDistance { get; private set; }
+        /// <remarks>
+        /// This is bounded by <see cref="JumpDistance"/>, but may be smaller if a more natural path is able to be taken through a preceding slider.
+        /// </remarks>
```

Comment:
I'm still not sure how to read this correctly. There's a mention of a "minimum distance from end position of previous object", but then how can that be anything *other* than `JumpDistance` if it's in a straight line?

The "end position" part I especially take issue with, because as far as I can tell from [the picture in the other PR thread correctly](https://github.com/ppy/osu/pull/15758#issuecomment-977468402) and some inline comments from this pull, the point is that a path with length of `MovementDistance` does *not* necessarily hit the end position of the previous object, it just moves the cursor enough to *not miss it*?

Given my understanding above is correct, only then it starts to makes sense to me that `JumpDistance` is the upper bound of `MovementDistance`. (I would also specify the "upper bound" part explicitly in this xmldoc.)

Reply:
I'm not sure where this review conversation is at after the latest changes - is there something I should recommend here? Because I'm admittedly still confused as to what `Jump` and `MovementDistance` are

Reply:
I haven't gotten around to looking into it yet. The pushed commit was supposed to be pushed quite a long time ago, it was just sitting in my local working set.